src/pages/projects/AddProjectContributorForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { Box, Button, MenuItem, TextField } from '@mui/material';
+import { useFormikContext } from 'formik';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { ContributorSearchField } from '../../components/ContributorSearchField';
+import { StyledRightAlignedFooter } from '../../components/styled/Wrappers';
+import { setNotification } from '../../redux/notificationSlice';
+import {
+  CristinProject,
+  ProjectContributor,
+  ProjectContributorType,
+  ProjectFieldName,
+} from '../../types/project.types';
+import { CristinPerson } from '../../types/user.types';
+import { dataTestId } from '../../utils/dataTestIds';
+
+const roles: ProjectContributorType[] = ['ProjectManager', 'ProjectParticipant'];
+
+interface AddProjectContributorFormProps {
+  hasProjectManager: boolean;
+  toggleModal: () => void;
+}
+
+export const AddProjectContributorForm = ({ hasProjectManager, toggleModal }: AddProjectContributorFormProps) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const { values, setFieldValue } = useFormikContext<CristinProject>();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [selectedContributorRole, setSelectedContributorRole] = useState(hasProjectManager ? roles[1] : roles[0]);
+  const [selectedPerson, setSelectedPerson] = useState<CristinPerson>();
+
+  const addContributor = () => {
+    if (!selectedPerson || !selectedContributorRole) {
+      return;
+    }
+
+    if (selectedContributorRole === 'ProjectManager' && hasProjectManager) {
+      dispatch(
+        setNotification({
+          message: t('project.error.there_can_only_be_one_project_manager_choose_different_role'),
+          variant: 'error',
+        })
+      );
+    }
+
+    const newContributor: ProjectContributor = {
+      identity: {
+        type: 'Person',
+        id: selectedPerson.id,
+        firstName: selectedPerson.names[0].type === 'FirstName' ? selectedPerson.names[0].value : '',
+        lastName: selectedPerson.names[1].type === 'LastName' ? selectedPerson.names[1].value : '',
```

Comment:
Vi gikk noen runder med hvordan modellen skal se ut for navn og sånt her, for jeg syns det er veldig tungvindt å jobbe med arrays med navn, i stedet for rene properties som `firstName:"xxx"` og `lastName:"yyy"`, men gikk ikke sånn... :/ 
Anyways; vi vet ikke om fornavn er i første eller fjerde indeks, så vi må faktisk lete gjennom hele lista for å være sikker.
```suggestion
        firstName: getValueByKey('FirstName', selectedPerson.names),
        lastName: getValueByKey('LastName', selectedPerson.names),
```

Lagde noen utility-metoder som endrer til objekter som er litt enklere å jobbe med, bl.a. `convertToFlatCristinPerson`. Men det er jo en ekstra overhead å caste frem og tilbake også da :(

Reply:
Fyi, så kan det være innslag for `preferredLastName`, og `preferredFirstName` i dette arrayet også. Mulig mer også, som jeg ikke vet om